src/main/scala/dpla/ingestion3/harvesters/Harvester.scala

```
@@ -0,0 +1,145 @@
+    */
+  protected def runHarvest: Try[DataFrame]
+
+  /**
+    * Abstract method mimeType should store the mimeType of the harvested data.
+    */
+  protected val mimeType: String
+
+  /**
+    * Entry point for running a harvest.
+    */
+  def harvest = {
+
+    // If the output directory already exists then delete it and its contents.
+    outputFile.getParentFile.mkdir()
+    if (outputFile.exists)
+      harvestLogger.info(s"Output directory already exists. Deleting ${outputDir}...")
+    Utils.deleteRecursively(outputFile)
+
+    // Log message that harvest is starting.
+    harvestLogger.info(s"Beginning ${provider} harvest")
+    val startTime = System.currentTimeMillis()
+
+    // Call local implementation.
+    runHarvest match {
+
+      case Success(df) =>
+        // Log details about the successful harvest.
+        val endTime = System.currentTimeMillis()
+        harvestLogger.info(s"Records saving to ${outputDir}")
+        Utils.logResults(endTime-startTime, df.count(), harvestLogger)
+        validateSchema(df)
+
+      case Failure(f) =>
+        // Log the failure.
+        harvestLogger.fatal(s"Unable to harvest records. ${f.getMessage}")
+    }
+
+    // Shut down spark session.
+    sc.stop()
+  }
+
+  /**
+    * The long name of the provider.
+    *
+    * @throws IllegalArgumentException if no provider specified in conf file.
+    */
+  protected val provider: String = conf.provider
+    .getOrElse(throw new IllegalArgumentException("No provider specified."))
+
+  protected val outputFile = new File(outputDir)
+
+  /**
+    * Initiate a spark session using the configs specified in the i3Conf.
+    *
+    * Lazy evaluation b/c some harvesters do not need a spark context until the
+    * very end of the harvest.
+    *
+    * @return SparkSession
+    */
+  protected lazy val spark: SparkSession = {
+    val sparkConf = new SparkConf().setAppName(s"Harvest: ${provider}")
+    sparkConf.setMaster(conf.spark.sparkMaster.get)
```

Should this be getOrElse("local[1]") or such? The conf file isn't doing any validation of its contents or do we want to throw an exception if values aren't specified in the conf file?

---

Yeah, that's a good idea.  I'll set a default.